woodwork/utils.py

```diff
@@ -27,14 +27,14 @@ def camel_to_snake(s):
     return re.sub('([a-z0-9])([A-Z])', r'\1_\2', s).lower()
 
 
-def _convert_input_to_set(semantic_tags, error_language='semantic_tags'):
+def _convert_input_to_set(semantic_tags, error_language='semantic_tags', validate=True):
     """Takes input as a single string, a list of strings, or a set of strings
         and returns a set with the supplied values. If no values are supplied,
         an empty set will be returned."""
     if not semantic_tags:
         return set()
 
-    if type(semantic_tags) not in [list, set, str]:
+    if validate and type(semantic_tags) not in [list, set, str]:
         raise TypeError(f"{error_language} must be a string, set or list")
```

Comment:
If we want to be able to use mocks to test whether this has been called, we'll want to pull this and the check below into their own function. 

It might be a bit excessive to pull these single lines into their own functions, and as we're not actually catching any errors when we don't validate and we just get unexpected results from this function, it might be so bad to not do that.

Anyway, I'd gone ahead to see what that would look like and was running into some unexpected behavior that I'll need to take another look at tomorrow to see if I can figure out if we do want to use mocks here as well.

Reply:
Turns out that in order to test exceptions that are raised by a function you've mocked, you need to specify the exceptions as a side effect. In the other tests where mocks have been added, we just test the non-error cases (for when we are validating), so instead of adding the side effect to a mocked `test_convert_input_to_set`, I just added a separate test to check validation control on `_convert_input_to_set`.